workers/unity/Assets/Playground/Scripts/Cubes/CubeMovementSystem.cs

```diff
@@ -43,17 +42,21 @@ protected override void OnUpdate()
             for (var i = 0; i < data.Length; i++)
             {
                 var rigidbodyComponent = data.Rigidbody[i];
-                if (rigidbodyComponent.position.x - worker.Origin.x > 10)
+                var cubeComponent = data.Cube[i];
+
+                if (cubeComponent.Velocity.X > 0 && rigidbodyComponent.position.x - worker.Origin.x > 10)
```

Comment:
Ideally get rid of magic numbers. Although this doesn't matter too much in playground.